core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/streams/streams.go

```diff
@@ -249,10 +253,12 @@ func (s *streams) DoMercuryRequest(ctx context.Context, lookup *mercury.StreamsL
 	pluginRetryKey := generatePluginRetryKey(checkResults[i].WorkID, lookup.Block)
 	upkeepType := core.GetUpkeepType(checkResults[i].UpkeepID)
 
+	reqCtx, cancel := context.WithTimeout(ctx, requestTimeout)
+	defer cancel()
 	if lookup.IsMercuryV02() {
-		state, values, errCode, retryable, retryInterval, err = s.v02Client.DoRequest(ctx, lookup, upkeepType, pluginRetryKey)
+		state, values, errCode, retryable, retryInterval, err = s.v02Client.DoRequest(reqCtx, lookup, upkeepType, pluginRetryKey)
```

Comment:
what happens when ctx ends here? We'll need to catch that and convert it to timeout error code

Reply:
Within `DoRequest()` the most problematic point is when making the actual request (`c.httpClient.Do()`) as we are waiting for response.
In that case we'll end up with:
```
retryable = true
state = encoding.MercuryFlakyFailure
errCode = encoding.ErrCodeStreamsUnknownError
```
We can set errCode to timeout in case the context is done, ensuring that timeout is reflected when needed

Reply:
Fixed